app/controllers/application_controller.rb

```diff
@@ -1,2 +1,38 @@
+require 'strata'
+require 'datadog/statsd'
+
 class ApplicationController < ActionController::Base
+  before_action :handle_subdomain
+
+  private
+
+  def handle_subdomain
+    @tenant = request.subdomain
```

Comment:
You don't need an instance variable `@tenant`
```suggestion
    if request.subdomain.present?
       Tenant.find_by(domain: request.subdomain).switch!
       # logger
     else
         restore_default_searchpath
     end
```